src/test/java/org/elasticsearch/index/query/TemplateQueryTest.java

```diff
@@ -371,6 +376,45 @@ public void testIndexedTemplate() throws Exception {
         assertHitCount(sr, 4);
     }
 
+    // Relates to #10397
+    @Test
+    public void testIndexedTemplateOverwrite() throws Exception {
+        createIndex("testindex");
+        ensureGreen("testindex");
+
+        index("testindex", "test", "1", jsonBuilder().startObject().field("searchtext", "dev1").endObject());
+        refresh();
+
+        PutIndexedScriptResponse scriptResponse = client().preparePutIndexedScript(MustacheScriptEngineService.NAME, "git01", 
```

Comment:
Maybe I'm confused but I think we are talking about two different templates:

The template this test is checking is the following: http://www.elastic.co/guide/en/elasticsearch/reference/1.4/search-template.html


The template that can be stored with the methods you mentioned seems to refer to the following:
http://www.elastic.co/guide/en/elasticsearch/reference/1.x/indices-templates.html